app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -294,6 +296,7 @@ async function addEthereumChainHandler(
         fromNetworkConfiguration: getProviderConfig(),
       },
     });
+    setNetworkClientIdForDomain(req.origin, networkConfigurationId);
```

Comment:
This fixes a bug where if a dapp prompts an add-ethereum-chain request and the user then accepts the subsequent `switchEthereumChain` call that is chained on, the `networkClientID` set for that dapp in the `SelectedNetworkController` domains state will not be updated.